### ISSUE TITLE
Mocking in tests?

### ISSUE THREAD

**Issue by billbrod** [flatironinstitute/ccn-template#42]
At US-RSE'23, heard several people describe the utility of "mocking" when writing tests, e.g., [mockito](https://mockito-python.readthedocs.io/en/latest/walk-through.html). Look into whether that's something we should use more.

**Comment by EricThomson** [comment]
Yes -- big topic. It might be nice to discuss the divisions among things like mocks and fixtures (e.g., tmp_path is a fixture baked into pytest), and tools things like freezegun (it provides specific dates dates/times for testing -- I found it helpful at one point but now it may be replaced by a standard built in for things like mockito which I'd never even heard of :scream:).  Rabbit hole alert!!
